applications/skyepass.md

```diff
@@ -0,0 +1,251 @@
+
+Some ideas we have had so far:
+
+- `Crypto Wallet`: shared hot wallet. The owner of the vault can install an `Ethereum` extension and store the private key with it. And, of course, DApp browsers. 
+- `SSH Login Tool`: a whole team can share login credential to their server effortlessly. 
+- `Shared Phone Number`: a shared Google account that registered on `Google Voice` can be stored, and the whole family can receive verification code for services. 
+
+
+
+### Password Manager & an Identity Management Solution
+
+Based on some thinking of the basis nature of NFTs. We believe that a password manager is an ideal medium to deliver tokenized digital identities. Therefore, we think each username-password-OTP combination as an atomic token, a vault as a collection of these identities, and an `extension` as a service injected with an identity. 
+
+- If we assume all identity tokens have two states: "public identity" or "private identity". A public handle is the public identity of a user. (i.e. a twitter handler, a Github handle or a Venmo handle etc. ) Therefore, we are building a solution to link to one's public off-chain profiles. Also, we can implement a ENS-like or `@username` style handle system.
+- Therefore, simple sharing behavior (i.e. share my spotify account to my girlfriend) can take two forms: if she has an account with this password manager, simple `@her`, set some rules for using this password(or not) and press share. If she has not, a one time sharing link will be sent, her browser will generate an ephemeral key pair, and that ephemeral key pair will be used to encrypted the entry and send the encrypted password entry over and make it self-destruct soon.
+- For teams or families, they are using a shared identity. They can link their profiles and get a handle like `@team`, while the team will use some secret sharing schema for privilege management.
+
+For more on this, please refer to `Future Plan/Integrated Identity Solution` section. 
+
+### Ecosystem Fit 
+I don't think there are anything like SkyePass so far, both within the Substrate community or all blockchain communities. The reason that Substrate will be an ideal platform for SkyePass is because the flexibility the framework offers. We plan to deploy 4 smart contracts:
+
+- Contract 1: A NFT contract that issues an NFT token for vault creator, store metadata of the vault and manage permission to update the vault. (the `vault metadata contract`)
+
+- Contract 2: A NFT contract that generate an NFT token that represents an atomic digital identity to users (the `atomic digital identity contract`)
+
+- Contract 3: A generalized handle system (an ENS-like system) on a Substrate-based chain that issue user handle ownership NFTs (the `substrate identity handle contract`)
+
+- Contract 4: An identity control smart contract that verify and store off-chain user handle(s), in compliment to the `substrate identity handle contract` (the `off-chain identity linkage contract`)
```

Comment:
What kind of "user handle verification" can a smart contract do?

Reply:
Link off-chain identity onto the chain. It's gonna work together with Contract 3... and we might combine them, or make them pallet instead of smart contracts, as mentioned in the "smart contract" part of the milestone deliverables. 

We took inspiration from the [Celo Network](https://celo.org/)'s phone number as public address mechanism. These contracts are planned with the intention to make sharing password easier. When a social media handle is linked to an on-chain identity, sharing a password is gonna be like sending money to someone on Venmo. 

Passwords and user handles are like `private keys` and `public keys`. Since, we are building a password manager that handles the `private keys` (i.e. account name and password) to users' offchain accounts credentials, it comes naturally for us to think if we can make a very cool experience for user to link and verify their offchain user handles. I guess, it can be an entirely new project to work on, but we do want to do at least some basic thinking into it on this product. 

Tbh, as of now, We haven't come to fully deep dive into it yet. As suggested by David, we are investigating the usage of the `identity` pallet, and call the pallet on the client side (so no smart contracts3 and 4); or build/use something like https://github.com/sunshine-protocol/sunshine-keybase. Again, we are on early stage for this question, I hope my answer gives you a rough idea of our thinking process. Let me know how you think.